src/gui/torrentoptionsdialog.h

```diff
@@ -26,40 +26,70 @@
  * exception statement from your version.
  */
 
-#ifndef UPDOWNRATIODIALOG_H
-#define UPDOWNRATIODIALOG_H
+#pragma once
 
 #include <QDialog>
 
+namespace BitTorrent
+{
+    class InfoHash;
+    class TorrentHandle;
+}
+
 namespace Ui
 {
-    class UpDownRatioDialog;
+    class TorrentOptionsDialog;
 }
 
-class UpDownRatioDialog final : public QDialog
+class TorrentOptionsDialog final : public QDialog
 {
     Q_OBJECT
 
 public:
-    UpDownRatioDialog(bool useDefault, qreal initialValue, qreal maxValue,
-            int initialTimeValue, int maxTimeValue,
-            QWidget *parent = nullptr);
-    ~UpDownRatioDialog();
-
-    bool useDefault() const;
-    qreal ratio() const;
-    int seedingTime() const;
+    explicit TorrentOptionsDialog(QWidget *parent, const QVector<BitTorrent::TorrentHandle *> &torrents);
+    ~TorrentOptionsDialog() override;
 
 public slots:
     void accept() override;
 
 private slots:
+    void updateUploadSpinValue(int val);
+    void updateDownloadSpinValue(int val);
+    void updateUploadSliderValue(int val);
+    void updateDownloadSliderValue(int val);
+
+    void handleUpSpeedLimitChanged();
+    void handleDownSpeedLimitChanged();
+    void handleDisableDHTChanged();
+    void handleDisablePEXChanged();
+    void handleDisableLSDChanged();
+
     void handleRatioTypeChanged();
-    void enableRatioSpin();
-    void enableTimeSpin();
 
 private:
-    Ui::UpDownRatioDialog *m_ui;
+    struct initialValues
```

Comment:
Please move this down to near `m_ui`.
Also you should rewrite it:
```c++
struct 
{
//  fields
} m_initialValues;
```

Reply:
Done.